### DISCUSSION TITLE
How models Identify Labels

### DISCUSSION THREAD

**Discussion by princeannan4332** [mrdbourke/pytorch-deep-learning#621] [General]
I am a self taught beginner in machine learning and I want to understand how models know that this index is the right choice corresponding to the label based on the prediction probabilities in a multiclass classification.Let say we are having images like beans->0 and rice->1 and wheat->3 (The numbers are the labels for each food image).Let say we insert a dataset of these images in  this order [beans,rice,rice,beans,beans,wheat].From visualization I know that the labels are [0,11,0,0,3] respectively.So please what I want to understand is that when we insert the data [beans,rice,rice,beans,beans,wheat] into the model and tell it to make 4 prediction probabilties as out_features.How will it know that this is the index that it is  suppose to place the highest prediction probability with respect to the actual label.Example:Let say we fed the model with a picture of rice.How will the model produce an outcome of [0.2,0.9,0.3,0.4] where the 0.9 is the highest probability predicted and its index is 1 which corresponds to the label of rice->1 and why didn't it put the highest prediction probabilty else where like [0.9,0.2,0.3,0.4] since the data  [beans,rice,rice,beans,beans,wheat] inserted was mixed up

**Comment by apollo000104** [comment]
Hello, @princeannan4332 
I hope to help you.
Can you explain what you want to know?
you want to know about how the model inference a probability list like [0.2, 0.9, 0.3, 0.4] , right?
it is just the machine learning model do.
In other words, the front part of model study the features of training data using one-hot-encoding. then the features pass to the softmax function to get a probability list what you mentioned.[0.2, 0.9, 0.3, 0.4]
hope to this helps you

**Reply by princeannan4332** [reply]
Thanks @apollo000104 😊

**Comment by mrdbourke** [comment]
Hi @princeannan4332 , 

Great question!

The main reason is that the model learns the relationship between the input data and the output label space.

So even though the order of data in the future may be different from what the model has seen before, it can adapt.

When it makes a prediction, it's always making a prediction of one sample in -> one prediction out (as long as that's how it's been trained, some models can make multiple predictions).

E.g. if their are 100 possible labels, for a given sample, the model will assign a score for each label, and in classification problems, the label with the highest score is classified as the prediciton.

**Reply by princeannan4332** [reply]
Thank you @mrdbourke for making me understand
